analyzers/src/SonarAnalyzer.CSharp/Rules/UnchangedLocalVariablesShouldBeConst.cs

```diff
@@ -147,31 +131,32 @@ private static bool HasMutableUsagesInMethod(VariableDeclaratorSyntax variable,
                 parentSyntax = parentSyntax.Parent;
             }
 
+            ISymbol variableSymbol = null;
+
             return parentSyntax
                 .DescendantNodes()
                 .OfType<IdentifierNameSyntax>()
                 .Where(MatchesIdentifier)
                 .Any(x => IsMutatingUse(semanticModel, x));
 
             static bool IsMethodLike(SyntaxNode arg) =>
-                arg is BaseMethodDeclarationSyntax
-                || arg is IndexerDeclarationSyntax
-                || arg is AccessorDeclarationSyntax
-                || arg is LambdaExpressionSyntax
-                || arg is GlobalStatementSyntax;
-
-            bool MatchesIdentifier(IdentifierNameSyntax id) =>
-                Equals(variableSymbol, semanticModel.GetSymbolInfo(id).Symbol);
+                arg is BaseMethodDeclarationSyntax or IndexerDeclarationSyntax or AccessorDeclarationSyntax or LambdaExpressionSyntax or GlobalStatementSyntax;
+
+            bool MatchesIdentifier(IdentifierNameSyntax id)
+            {
+                variableSymbol ??= semanticModel.GetDeclaredSymbol(variable);
+                return variableSymbol.Equals(semanticModel.GetSymbolInfo(id).Symbol);
```

Comment:
I think that `GetDeclaredSymbol` can return null, right?  should do a null check before calling Equals?